ovs/datapath_test.go

```diff
@@ -0,0 +1,644 @@
+				},
+			},
+			zone:   specificZone,
+			dpName: "system@ovs-system",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			res, err := tt.dp.SetCTLimits(tt.dpName, tt.zone)
+			if err != nil {
+				t.Errorf("got %q and %q", res, err.Error())
+			}
+		})
+	}
+}
+
+func TestSetCTLimitsWithBinary(t *testing.T) {
+	if testing.Short() {
+		// if you want to run tests in a context where
+		// you don't get ovs-dpctl installed, you should run
+		// go test -short to skip it
+		t.Skip("skipping test in short mode.")
+	}
+	defaultZone := make(map[string]uint64)
+	defaultZone["default"] = 200
+	specificZone := make(map[string]uint64)
+	specificZone["zone"] = 4
+	specificZone["limit"] = 4000
+
+	var tests = []struct {
+		desc   string
+		dp     *DataPathService
+		zone   map[string]uint64
+		dpName string
+		want   string
+		err    string
+	}{
+		{
+			desc:   "Test parse valid default argument",
+			dp:     NewDataPathService(),
+			zone:   defaultZone,
+			dpName: "system@ovs-system",
+		},
+		{
+			desc:   "Test parse valid default argument",
+			dp:     NewDataPathService(),
+			zone:   specificZone,
+			dpName: "system@ovs-system",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			res, err := tt.dp.SetCTLimits(tt.dpName, tt.zone)
+			if err != nil {
+				t.Errorf("got %q and %q", res, err.Error())
+			}
+		})
+	}
+}
+
+func TestDelCTLimits(t *testing.T) {
+	var tests = []struct {
+		desc   string
+		dp     *DataPathService
+		zones  []uint64
+		dpName string
+		err    string
+		res    int
+	}{
+		{
+			desc: "Test del limit with valid argument",
+			dp: &DataPathService{
+				CLI: &MockOvsCLI{
+					Version: 10,
+				},
+			},
+			zones:  []uint64{4, 3},
+			dpName: "system@ovs-system",
+			res:    0,
+		},
+		{
+			desc: "Test del limit with datapath missing",
+			dp: &DataPathService{
+				CLI: &MockOvsCLI{
+					Version: 10,
+				},
+			},
+			zones:  []uint64{4, 3},
+			dpName: "",
+			err:    "datapath name is missing",
+			res:    1,
+		},
+		{
+			desc:   "Test del limit with empty paramaters",
+			dp:     NewDataPathService(),
+			zones:  []uint64{},
+			dpName: "",
+			err:    "datapath name is missing",
+			res:    1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			res, err := tt.dp.DelCTLimits(tt.dpName, tt.zones)
+			switch tt.res {
+			case 0:
+				if err != nil {
+					t.Errorf("got %q and %q", res, err.Error())
+				}
+
+			case 1:
+				if err.Error() != tt.err {
+					t.Errorf("got %q and %q", res, err.Error())
+				}
+			}
+		})
+	}
+}
+
+func TestDelCTLimitsWithBinary(t *testing.T) {
+	if testing.Short() {
+		// if you want to run tests in a context where
+		// you don't get ovs-dpctl installed, you should run
+		// go test -short to skip it
+		t.Skip("skipping test in short mode.")
+	}
+
+	var tests = []struct {
+		desc   string
+		dp     *DataPathService
+		zones  []uint64
+		dpName string
+		err    string
+		res    int
```

Comment:
what is the purpose of this variable ?

Reply:
Oh, my bad it is lack of consistency I should have called it `testCase`.  But it is used to switch inside the `tt`